services/core/PlatformDriverAgent/platform_driver/interfaces/home_assistant.py

```diff
@@ -0,0 +1,308 @@
+# }}}
+
+
+import random
+from math import pi
+import json
+import sys
+from platform_driver.interfaces import BaseInterface, BaseRegister, BasicRevert
+from volttron.platform.agent import utils #added this to pull from config store 
+from volttron.platform.vip.agent import Agent
+import logging
+import requests
+from requests import get
+from collections.abc import MutableMapping
+
+_log = logging.getLogger(__name__)
+type_mapping = {"string": str,
+                "int": int,
+                "integer": int,
+                "float": float,
+                "bool": bool,
+                "boolean": bool}
+
+# def flatten_dict(d: MutableMapping, parent_key: str = '', sep: str ='.') -> MutableMapping:
+#     items = []
+#     for k, v in d.items():
+#         new_key = parent_key + sep + k if parent_key else k
+#         if isinstance(v, MutableMapping):
+#             items.extend(flatten_dict(v, new_key, sep=sep).items())
+#         else:
+#             items.append((new_key, v))
+#     return dict(items)
+
+class HomeAssistantRegister(BaseRegister):
+    def __init__(self, read_only, pointName, units, reg_type, attributes,
+                 default_value=None, description=''):
+        super(HomeAssistantRegister, self).__init__("byte", read_only, pointName, units,
+                                           description='')
+        self.reg_type = reg_type
+        self.attributes = attributes
+
+        if default_value is None:
+            self.value = self.reg_type(random.uniform(0, 100))
+        else:
+            try:
+                self.value = self.reg_type(default_value)
+            except ValueError:
+                self.value = self.reg_type()
+
+
+class Interface(BasicRevert, BaseInterface):
+    def __init__(self, **kwargs):
+        super(Interface, self).__init__(**kwargs)
+        self.point_name = None
+  
+    def configure(self, config_dict, registry_config_str): # grabbing from config
+        self.ip_address = config_dict.get("ip_address", "0.0.0.0")
+        self.access_token = config_dict.get("access_token", "access_token")
+        self.volttron_topic = config_dict.get("volttron_topic", "devices")
+        self.points_to_grab_from_topic = config_dict.get("points_to_grab_from_topic", "points_to_grab_from_topic")
+        self.port = config_dict.get("port", "port")
+        self.registry_config = config_dict.get("registry_config","registry_config")
+        self.parse_config(registry_config_str) 
+        
+    def get_point(self, point_name):
+        register = self.get_register_by_name(point_name)
+        return register.value
+
+    def _set_point(self, point_name, value):
+        register = self.get_register_by_name(point_name)
+        if register.read_only:
+            raise RuntimeError(
+                "Trying to write to a point configured read only: " + point_name)
+        register.value = register.reg_type(value) # setting the value
+
+
+        if register.value == True:
```

Comment:
How do you know True/False is turning lights on or off?

Reply:
Because it is setup as a boolean. If its changed to true it calls the function to turn on that light. I guess now that I think about it there is no real correlation between that and Home Assistant but it works for now.